src/DataStructures/LeviCivitaIterator.hpp

```diff
@@ -61,21 +62,78 @@ class LeviCivitaIterator {
   }
 
   // Return the current (multi-index) Index, an ordered list of indices
-  const Index<Dim>& operator()() const noexcept {
+  const cpp17::array<size_t, Dim>& operator()() const noexcept {
```

Comment:
Why are `cpp17::array`s being preferred to `std::array`s in this PR?  This is just going to confuse people and make then convert back and forth when they want to use this along with other functions.

If there is a good reason please document it, at least in a commit message.

Reply:
Because std::array is not contexpr, so I can't use std::array if I want to constexpr the variables I was asked to constexpr below. cpp17::array is constexpr, so I can use it in constexpr functions.